src/Infrastructure.Styles.Analyzer/RMSYN1104OperatorChainingWithNewlineAnalyzer.cs

```diff
@@ -0,0 +1,137 @@
+          SyntaxKind.GreaterThanExpression,
+          SyntaxKind.EqualsExpression,
+          SyntaxKind.NotEqualsExpression);
+
+      context.RegisterSyntaxNodeAction(AnalyzeBinaryExpressionWithToken,
+          SyntaxKind.LogicalOrExpression,
+          SyntaxKind.LogicalAndExpression,
+          SyntaxKind.BitwiseOrExpression,
+          SyntaxKind.BitwiseAndExpression,
+          SyntaxKind.ExclusiveOrExpression);
+
+      context.RegisterSyntaxNodeAction(AnalyzeBinaryExpressionWithToken,
+          SyntaxKind.IsExpression);
+
+      context.RegisterSyntaxNodeAction(AnalyzeNonBinaryExpressionWithTokens,
+          SyntaxKind.IsPatternExpression,
+          SyntaxKind.OrPattern,
+          SyntaxKind.AndPattern);
+
+      // ? part of ?. operator
+      context.RegisterSyntaxNodeAction(AnalyzeConditionalAccessExpression, SyntaxKind.ConditionalAccessExpression);
+      // . part of ?. operator
+      context.RegisterSyntaxNodeAction(AnalyzeNonBinaryExpressionWithTokens, SyntaxKind.MemberBindingExpression);
+
+      // ??
+      context.RegisterSyntaxNodeAction(AnalyzeBinaryExpressionWithToken, SyntaxKind.CoalesceExpression);
+
+      // .ToString().Whatever().UndSoWeiter
+      context.RegisterSyntaxNodeAction(AnalyzeMemberAccessExpression, SyntaxKind.SimpleMemberAccessExpression);
+
+      //ternary operator, aka: something ? yes : no
+      context.RegisterSyntaxNodeAction(AnalyzeNonBinaryExpressionWithTokens, SyntaxKind.ConditionalExpression);
+    }
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+    // Fürs review: hier haben wir 3 mal basically die gleiche Methode, das kann man glaube ich irgendwie reduzieren, 
+    // ich weiß allerdings nicht wie. Pls advise me :)
+    // Nur infomäßig, BinaryExpression, MemberAcessExpression etc. erben alle von expression, die property wird allerdings
+    // lokal deklariert
+    private static void AnalyzeBinaryExpressionWithToken (SyntaxNodeAnalysisContext context)
+    {
+      var expression = (BinaryExpressionSyntax)context.Node;
+      ReportDiagnosticIfContainsTrailingEndLine(expression.OperatorToken, context);
+    }
+
+    private static void AnalyzeMemberAccessExpression (SyntaxNodeAnalysisContext context)
+    {
+      var expression = (MemberAccessExpressionSyntax)context.Node;
+      ReportDiagnosticIfContainsTrailingEndLine(expression.OperatorToken, context);
+    }
+
+    private static void AnalyzeConditionalAccessExpression (SyntaxNodeAnalysisContext context)
+    {
+      var expression = (ConditionalAccessExpressionSyntax)context.Node;
+      ReportDiagnosticIfContainsTrailingEndLine(expression.OperatorToken, context);
+    }
+
+    private static void AnalyzeNonBinaryExpressionWithTokens (SyntaxNodeAnalysisContext context)
+    {
+      foreach (var invalidChildToken in context.Node.ChildTokens()) ReportDiagnosticIfContainsTrailingEndLine(invalidChildToken, context);
```

Comment:
`.ChildTokens()` creates an enumerable to iterate over all the child tokens. We should always just do a an expression case and property access. To prevent any performance issues. (This should be pretty straightforward with the generic implementation so maybe we should to it).